### PR TITLE
TEST: fixed compilation by gcc 7.1.1

### PR DESCRIPTION
**Pull request by hoopoepg**
- fixed potential issue reported by gcc 7.1.1
- added configuration scripts to skip tests on
  malloc state API is unavailable

this is fix for issue: https://github.com/openucx/ucx/issues/1779

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4533/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by hoopoepg** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4536/ for details (Mellanox internal link).

File: config/m4/ucm.m4 [hunk 1]
```diff
@@ -48,3 +48,8 @@ esac
 AM_CONDITIONAL([HAVE_UCM_PTMALLOC283],[test "x$HAVE_UCM_PTMALLOC283" = "xyes"])
 AM_CONDITIONAL([HAVE_UCM_PTMALLOC286],[test "x$HAVE_UCM_PTMALLOC286" = "xyes"])
 
+AC_CHECK_FUNCS([malloc_get_state malloc_set_state],
+               [],
+               [],
+               [#include <stdlib.h>])
```
Comment (yosefe) [inline config/m4/ucm.m4]: shouldn't this be `#include <malloc.h>`?

File: test/gtest/ucm/malloc_hook.cc [hunk 2]
```diff
@@ -56,6 +61,7 @@ class malloc_hook : public ucs::test {
 
 int malloc_hook::small_alloc_count = 1000 / ucs::test_time_multiplier();
 
+#if HAVE_MALLOC_STATES
```
Comment (yosefe) [inline test/gtest/ucm/malloc_hook.cc]: can we just skip the test part which calls malloc_get_state/malloc_set_state? wouldn't want to disable malloc hooks tests completely

File: test/gtest/ucs/test_log.cc [hunk 3]
```diff
@@ -12,7 +12,7 @@ class log_test : public ucs::test {
 
 public:
     virtual void init() {
-        char ucs_log_spec[64];
+        char ucs_log_spec[70];
```
Comment (yosefe) [inline test/gtest/ucs/test_log.cc]: why is this needed?

File: config/m4/ucm.m4 (hunk 1 above)
Comment (hoopoepg) [reply config/m4/ucm.m4]: this is fix for GCC compilation warning:
```
ucs/test_log.cc:14:18: error: ‘%s’ directive output may be truncated writing up to 63 bytes into a region of size 59 [-Werror=format-truncation=]
     virtual void init() {
                  ^~~~
ucs/test_log.cc:29:17: note: ‘snprintf’ output between 6 and 69 bytes into a destination of size 64
         snprintf(ucs_log_spec, sizeof(ucs_log_spec), "file:%s", logfile);
         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: test/gtest/ucs/test_log.cc (hunk 3 above)
Comment (hoopoepg) [reply test/gtest/ucs/test_log.cc]: this is fix for GCC compilation warning:
```
ucs/test_log.cc:14:18: error: ‘%s’ directive output may be truncated writing up to 63 bytes into a region of size 59 [-Werror=format-truncation=]
     virtual void init() {
                  ^~~~
ucs/test_log.cc:29:17: note: ‘snprintf’ output between 6 and 69 bytes into a destination of size 64
         snprintf(ucs_log_spec, sizeof(ucs_log_spec), "file:%s", logfile);
         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: config/m4/ucm.m4 (hunk 1 above)
Comment (hoopoepg) [reply config/m4/ucm.m4]: according man pager stdlib.h should be used

File: test/gtest/ucm/malloc_hook.cc (hunk 2 above)
Comment (hoopoepg) [reply test/gtest/ucm/malloc_hook.cc]: done

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4547/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:bgate:retest

**Comment by hoopoepg** [comment]
fixes https://github.com/openucx/ucx/issues/1779
